VMEncryption/main/EncryptionSettingsUtil.py

```diff
@@ -182,30 +184,52 @@ def _dict_to_name_value_array(self, values):
                     })
         return array
 
-    def get_settings_data(self, protector_name, kv_url, kv_id, kek_url, kek_kv_id, kek_algorithm, extra_device_items, disk_util, crypt_mount_config_util):
+    def get_settings_data(self, protector_name, kv_url, kv_id, kek_url, kek_kv_id, kek_algorithm, extra_device_items, disk_util, crypt_mount_config_util, key_store_type, keystoretype_flag_exists):
         """ returns encryption settings object in format required by wire server """
 
-        # validate key vault parameters prior to creating the encryption settings object
         cutil = CheckUtil(self.logger)
-        cutil.check_kv_id(kv_id, "A KeyVault ID is required, but is missing or invalid")
-        cutil.check_kv_url(kv_url, "A KeyVault URL is required, but is missing or invalid")
-        cutil.check_kv_name(kv_id, kv_url, "A KeyVault ID and KeyVault URL were provided, but their key vault names did not match")
-        if kek_url:
-            cutil.check_kv_id(kek_kv_id, "A KEK URL was specified, but its KEK KeyVault ID was missing or invalid")
-            cutil.check_kek_url(kek_url, "A KEK URL was specified, but it was invalid")
-            cutil.check_kek_name(kek_kv_id, kek_url, "A KEK ID and KEK URL were provided, but their key vault names did not match")
-            if kek_algorithm not in CommonVariables.encryption_algorithms:
-                if kek_algorithm:
-                    raise Exception("The KEK encryption algorithm requested was not recognized")
-                else:
-                    kek_algorithm = CommonVariables.default_encryption_algorithm
-                    self.logger.log("No KEK algorithm specified, defaulting to {0}".format(kek_algorithm))
+
+        # Perform algorithm check regaurdless of ManagedHSM or Keyvault
+        if kek_algorithm not in CommonVariables.encryption_algorithms:
+                    if kek_algorithm:
+                        raise Exception("The KEK encryption algorithm requested was not recognized")
+                    else:
+                        kek_algorithm = CommonVariables.default_encryption_algorithm
+                        self.logger.log("No KEK algorithm specified, defaulting to {0}".format(kek_algorithm))
+
+        # ManagedHSM Checks
+        if key_store_type and key_store_type.lower() == CommonVariables.KeyStoreTypeManagedHSM.lower():
```

Comment:
Not required to change for this PR but are these change redundant? There seems to be similar check happening in validate_key_vault_param in check_util.

Reply:
It does seem like it is, and was already, redundant before my updates. When we remove the KeyStoreType flag for public MHSM announcement, I will refactor this to be cleaner, as the KV parameters are also taken care of in the validate_key_vault_param, which is called in a precheck for fatal errors.